mmengine/utils/progressbar_rich.py

```diff
@@ -0,0 +1,365 @@
+                    f' {fps:.1f} tasks/s', self.colors[task_id])
+        else:
+            assert task_id >= 0 and task_id < len(self.tasks)
+            self.bar.update(self.tasks[task_id], advance=advance)
+            time.sleep(0.01)
+
+            if self.bar.finished:
+                self.bar.stop()
+
+    def reset(self):
+        for i in range(len(self.tasks)):
+            self.bar.remove_task(self.tasks[i])
+        self.tasks.clear()
+        self.colors.clear()
+
+
+def track_single_progress(func,
+                          tasks,
+                          description='Process...',
+                          color='blue',
+                          **kwargs):
+    """Track the progress of tasks execution with a progress bar.
+
+    Tasks are done with a simple for-loop.
+
+    Args:
+        func (callable): The function to be applied to each task.
+        tasks (list or tuple[Iterable, int]): A list of tasks or
+            (tasks, total num).
+        description (str): The description of progress bar.
+        color (str): The color of progress bar.
+
+    Returns:
+        list: The task results.
+    """
+    if isinstance(tasks, tuple):
+        assert len(tasks) == 2
+        assert isinstance(tasks[0], Iterable)
+        assert isinstance(tasks[1], int)
+        task_num = tasks[1]
+        tasks = tasks[0]
+    elif isinstance(tasks, Iterable):
+        task_num = len(tasks)
+    else:
+        raise TypeError(
+            '"tasks" must be an iterable object or a (iterator, int) tuple')
+    prog_bar = RichProgressBar()
+    prog_bar.add_single_task(task_num, color=color, description=description)
+    results = []
+    for task in tasks:
+        results.append(func(task, **kwargs))
+        prog_bar.update()
+    return results
+
+
+def track_multi_progress(funcs, tasks, descriptions, colors, params=None):
+    """Track multi progress of tasks execution with progress bar.
+
+    Tasks are done with a while-loop and a simple for-loop.
+
+    Args:
+        funcs (list): Functions apply for each task.
+        tasks (list): A list of tasks.
+        descriptions (str or list): The descriptions of each progress bar.
+        colors (str or list): The colors of each progress bar.
+        params (list): The funcs` parameters.
+
+    Returns:
+        list: The task results.
+    """
+    assert isinstance(funcs, List)
+    assert isinstance(tasks, List)
+    if params is None:
+        params = [dict() for i in range(len(funcs))]
+
+    if isinstance(descriptions, str):
+        descriptions = [descriptions] * len(tasks)
+    if isinstance(colors, str):
+        colors = [colors] * len(tasks)
+
+    assert len(funcs) == len(tasks) == len(params) == len(descriptions) == len(
+        colors)
+
+    prog_bar = RichProgressBar()
+    for i in range(len(tasks)):
+        total = len(tasks[i])
+        prog_bar.add_multi_task(total, colors[i], descriptions[i])
+
+    result = [[] for i in range(len(tasks))]
+    idx = 0
```

Comment:
What does `idx` mean here? BTW, is it necessary to provide the `track_multi_progress` here? It seems a little bit complex. Will it be more convenient for users to write a for loop to call `track_single_progress`?